crates/ethereum-engine/src/main.rs

```diff
@@ -121,17 +108,13 @@ pub fn main() {
         set_app_env(&config, &mut app, &path, path.len());
     }
     let matches = app.clone().get_matches();
-    match matches.subcommand() {
-        #[cfg(feature = "ethereum")]
-        ("ethereum-ledger", Some(ethereum_ledger_matches)) => {
-            merge_args(&mut config, &ethereum_ledger_matches);
-            tokio_run(run_ethereum_engine(
-                config.try_into::<EthereumLedgerOpt>().unwrap(),
-            ));
-        }
-        ("", None) => app.print_help().unwrap(),
-        _ => unreachable!(),
-    }
+    merge_args(&mut config, &matches);
+    // TODO: Abstract this over the backend being used.
+    #[cfg(feature = "redis")]
+    tokio_run(run_ethereum_engine(
+        config.try_into::<EthereumLedgerOpt>().unwrap(),
+    ));
+    unreachable!();
```

Comment:
i don't like this too much, but ideally with multiple backends this would be replaced by the other backends